File: test/language/subtypes/cpp98/TemplateArgumentSubtypeTest.cpp
#include "gtest/gtest.h"

#include "subtypes/template_argument_subtype/TemplateArgumentStructure.h"

namespace subtypes
{
namespace template_argument_subtype
{

TEST(TemplateArgumentSubtypeTest, testSubtype)
{
    Field_uint32 field_uint32;
    field_uint32.setValue(10);

    Compound compound;
    Field_Compound field_compound;
    field_compound.setValue(compound);

    TemplateArgumentStructure templateArgumentStructure;
    templateArgumentStructure.setAnotherUint32TypeField(field_uint32);
    templateArgumentStructure.setUint32TypeField(field_uint32);
    templateArgumentStructure.setUint32Field(field_uint32);
    templateArgumentStructure.setAnotherCompoundTypeField(field_compound);
    templateArgumentStructure.setCompoundTypeField(field_compound);
    templateArgumentStructure.setCompoundField(field_compound);
    ASSERT_EQ(192, templateArgumentStructure.bitSizeOf());
}

} // namespace template_argument_subtype
} // namespace subtypes

File: test/language/subtypes/java/subtypes/TemplateArgumentSubtypeTest.java
package subtypes;

import static org.junit.Assert.*;

import org.junit.Test;

import subtypes.template_argument_subtype.Field_uint32;
import subtypes.template_argument_subtype.Compound;
import subtypes.template_argument_subtype.Field_Compound;
import subtypes.template_argument_subtype.TemplateArgumentStructure;

public class TemplateArgumentSubtypeTest
{
    @Test
    public void testSubtype()
    {
        final Field_uint32 field_uint32 = new Field_uint32(10);
        final Field_Compound field_compound = new Field_Compound(new Compound(10));
        final TemplateArgumentStructure templateArgumentStructure = new TemplateArgumentStructure(field_uint32,
                field_uint32, field_uint32, field_compound, field_compound, field_compound);
        assertEquals(192, templateArgumentStructure.bitSizeOf());
    }
}
